nautilus_core/portfolio/src/portfolio.rs

```diff
@@ -54,9 +159,128 @@ pub struct Portfolio {
 }
 
 impl Portfolio {
-    // pub fn set_specific_venue(&mut self, venue: Venue) { // Lets try not to use this?
-    //     todo!()
-    // }
+    pub fn new(
+        msgbus: Rc<RefCell<MessageBus>>,
+        cache: Rc<RefCell<Cache>>,
+        clock: Rc<RefCell<dyn Clock>>,
+    ) -> Self {
+        let mut analyzer = PortfolioAnalyzer::new();
+
+        // Register default statistics
+        analyzer.register_statistic(Arc::new(MaxWinner {}));
+        analyzer.register_statistic(Arc::new(AvgWinner {}));
+        analyzer.register_statistic(Arc::new(MinWinner {}));
+        analyzer.register_statistic(Arc::new(MinLoser {}));
+        analyzer.register_statistic(Arc::new(MaxLoser {}));
+        analyzer.register_statistic(Arc::new(Expectancy {}));
+        analyzer.register_statistic(Arc::new(WinRate {}));
+        analyzer.register_statistic(Arc::new(ReturnsVolatility::new(None)));
+        analyzer.register_statistic(Arc::new(ReturnsAverage {}));
+        analyzer.register_statistic(Arc::new(ReturnsAverageLoss {}));
+        analyzer.register_statistic(Arc::new(ReturnsAverageWin {}));
+        analyzer.register_statistic(Arc::new(SharpeRatio::new(None)));
+        analyzer.register_statistic(Arc::new(SortinoRatio::new(None)));
+        analyzer.register_statistic(Arc::new(ProfitFactor {}));
+        analyzer.register_statistic(Arc::new(RiskReturnRatio {}));
+        analyzer.register_statistic(Arc::new(LongRatio::new(None)));
+
+        let portfolio = Self {
+            clock: clock.clone(),
+            cache: cache.clone(),
+            msgbus: msgbus.clone(),
+            accounts: AccountsManager::new(clock, cache.clone()),
+            analyzer,
+            unrealized_pnls: HashMap::new(),
+            realized_pnls: HashMap::new(),
+            net_positions: HashMap::new(),
+            pending_calcs: HashSet::new(),
+            initialized: false,
+        };
+
+        let portfolio_rc = Rc::new(RefCell::new(portfolio));
+        Self::register_message_handlers(&msgbus, &cache, &portfolio_rc);
+
+        Rc::try_unwrap(portfolio_rc)
+            .map_err(|_| "Failed to unwrap Portfolio")
+            .unwrap()
+            .into_inner()
+    }
+
+    fn register_message_handlers(
+        msgbus: &Rc<RefCell<MessageBus>>,
+        cache: &Rc<RefCell<Cache>>,
+        portfolio: &Rc<RefCell<Self>>,
```

Comment:
We cannot create an rc refcell of self from inside the new function. We will have to use an inner outer pattern here.

Also no need to pass rc refcell by & reference. Pass it by cloning it.